### PR TITLE
Issue-25: Support additional properties

### PR DESCRIPTION
**Pull request by EdwardSkoviak**
[test evidence](https://gist.github.com/EdwardSkoviak/8f179c03a2167ba854041921f79c47ed)

### CODE REVIEW

**Comment by EdwardSkoviak** [comment]
added ignore for FC022 (food critic rule 22) to the remote_file, cookbook_file, and template blocks in the provider, since the rule is not applicable to our only_if/not_if guards

verified this by adding the an additional true not_if guard first below, and then also above, the present false not_if for my_remote_file and got the following in both cases:

> - remote_file[/opt/my_dir/my_tomcat/my_remote_file] action create (skipped due to not_if)

**Comment by EdwardSkoviak** [comment]
Added additional test to the suite

**Comment by EdwardSkoviak** [comment]
[fresh test evidence](https://gist.github.cerner.com/ES028144/ceb2a46067317818d03be377b55aecc8)

File: README.md [hunk 1]
```diff
@@ -100,12 +100,17 @@ Parameters:
...
+ * `only_if`: Use a chef only_if [guard](https://docs.chef.io/resource_common.html#guards] to only write the cookbook file if the statement evaluates to true (optional).
+ * `not_if`: Use a chef not_if [guard](https://docs.chef.io/resource_common.html#guards] to not write the cookbook file if the statement evaluates to true (optional).
 
 Example:
 ``` ruby
 cerner_tomcat "my_tomcat" do
   cookbook_file "lib/mysql.jar" do
     source "mysql-5.12.jar"
+    only_if { true }
+    not_if { false }
```
Comment (cduranleau) [inline README.md]: Mind putting a slightly more complex example in here? Maybe something like File.exists?('mysql-5.12.jar')?

**Comment by cduranleau** [comment]
+1

**Comment by dhondgepooja** [comment]
+1

**Comment by mkwhitacre** [comment]
+1

File: README.md (hunk 1 above)
Comment (EdwardSkoviak) [reply README.md]: done: https://github.com/cerner/cerner_tomcat/pull/28/commits/074e83d1e75ac8cbd490bf704d6ef1b7db563601

**Comment by asandwich** [comment]
+1

**Comment by cduranleau** [comment]
+1

**Comment by EdwardSkoviak** [comment]
thanks for the reviews, going to squash this and merge

**Pull request merged**